src/extract/serde.rs

```diff
@@ -140,16 +145,20 @@ where T: DeserializeOwned,
                         .map_err(|_| Error::internal_error());
 
                     let res = try_ready!(res);
-
-                    // And here we deserialize, but we have not determined the
-                    // content type yet :(
-                    //
-                    // TODO: Make content type pluggable
-                    ::serde_json::from_slice(&res[..])
-                        .map_err(|_| {
-                            // TODO: Handle error better
-                            Some(Error::internal_error())
+                    
+                    if self.is_json == true {
+                        ::serde_json::from_slice(&res[..])
+                            .map_err(|_| {
+                                // TODO: Handle error better
+                                Some(Error::internal_error())
+                        })
+                    } else {
+                        serde_urlencoded::from_bytes(&res[..])
```

Comment:
Should this be `::serde_urlencoded` to match `::serde_json` above?

Reply:
If it compiles 🤷‍♂️ style will need to be cleaned up later anyway.